### PR TITLE
detail using cylc message to add to scheduler log

### PR DESCRIPTION
**Pull request by wxtim**
**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.

Locally viewable at http://tims-server/cylc-doc/8.1.3.dev/html/index.html

### CODE REVIEW

File: src/user-guide/running-workflows/scheduler-log-files.rst [hunk 1]
```diff
@@ -24,11 +24,50 @@ The information logged here includes:
...
+
+   some_function
+   cylc message -- "${CYLC_WORKFLOW_ID}" "${CYLC_TASK_JOB}" \
+       "ERROR:some_function failed."
+
+Your log should produce output similar to:
+
+.. code-block:: shell
+
+   ERROR - [21491012T0410Z/mytask running job:01 flows:1] (received)some_function failed ERROR at 2023-04-14T11:36:35+01:00
```
Comment (oliver-sanders) [inline src/user-guide/running-workflows/scheduler-log-files.rst:64]:
```suggestion
.. code-block:: shell

   cylc message -- "ERROR:some_function failed."

Your log should produce output similar to:

.. code-block:: none

   ERROR - [21491012T0410Z/mytask running job:01 flows:1] (received)some_function failed ERROR at 2023-04-14T11:36:35+01:00
```

In task environments, Cylc automatically loads the context from environment variables, example here:

https://github.com/cylc/cylc-flow/blob/master/tests/functional/restart/53-task-prerequisites/flow.cylc

File: src/user-guide/running-workflows/scheduler-log-files.rst [hunk 2]
```diff
@@ -24,11 +24,50 @@ The information logged here includes:
...
+
+.. _scheduler Logs.Cylc message:
+
+Cylc Message
+^^^^^^^^^^^^
+
+.. warning::
+
+   Using Cylc Message to add information to the scheduler logs should
+   be used sparingly to avoid overwhealming users with output.
```
Comment (oliver-sanders) [inline src/user-guide/running-workflows/scheduler-log-files.rst:45]: A warning makes this look scary. I'd be inclined to let users decide what level of logging makes sense and scratch these two lines.

File: src/user-guide/running-workflows/scheduler-log-files.rst [hunk 3]
```diff
@@ -24,11 +24,44 @@ The information logged here includes:
...
+
    Rather than parse the log files.
+
+
+.. _scheduler Logs.Cylc message:
+
+Cylc Message
+^^^^^^^^^^^^
+
+The Scheduler Log also records messages sent by ``cylc message`` allowing users
```
Comment (oliver-sanders) [inline src/user-guide/running-workflows/scheduler-log-files.rst:42]:
```suggestion
The Scheduler Log also records messages sent by ``cylc message`` allowing you
```

File: src/tutorial/furthertopics/message-triggers.rst [hunk 4]
```diff
@@ -51,6 +51,13 @@ In order to get our workflow to trigger messages, we need to:
 
 * Refer to these messages in the ``[dependencies]`` section of our workflow.
 
+.. note::
+
+   Messages are recorded in the Workflow's scheduler log. Make sure that
+   they convey useful information for users reading the log file. See
+   :ref:`scheduler logs.cylc message` for details of how messages appear in
+   the log file.
+
```
Comment (oliver-sanders) [inline src/tutorial/furthertopics/message-triggers.rst:60]:
```suggestion
.. note::

   The message will be recorded in the workflow's scheduler log.
   See :ref:`scheduler logs.cylc message` for details of how messages appear.

```

**Review by oliver-sanders — APPROVED** [review]

**Review by hjoliver — APPROVED** [review]
Good 👍

**Pull request merged**